ddmrp/models/product_product.py

```diff
@@ -11,16 +11,24 @@ class Product(models.Model):
         comodel_name="stock.buffer", string="Stock Buffers", inverse_name="product_id",
     )
 
-    def write(self, values):
-        res = super().write(values)
-        if "active" in values:
-            buffers = self.env["stock.buffer"].search(
+    archived_buffer_warning_qty = fields.Integer(
+        compute="_compute_archived_buffer_warning_qty", compute_sudo=True
+    )
+
+    @api.depends("buffer_ids", "buffer_ids.active")
+    def _compute_archived_buffer_warning_qty(self):
+        for product in self:
+            product.archived_buffer_warning_qty = self.env["stock.buffer"].search_count(
                 [
-                    ("product_id", "in", self.ids),
-                    "|",
-                    ("active", "=", True),
+                    ("product_id", "=", product.id),
+                    ("item_type", "=", "purchased"),
                     ("active", "=", False),
                 ]
             )
-            buffers.write({"active": values.get("active")})
+
+    def write(self, values):
+        res = super().write(values)
+        if values.get("active") is False:
+            buffers = self.env["stock.buffer"].search([("product_id", "in", self.ids)])
+            buffers.write({"active": False})
```

Comment:
buffer.active field could be related through product_id

Reply:
There can be many buffers for one product and some of them can be archived when the product is still active. So that would not work.